### ISSUE TITLE
Help on Handlebars Helper

### ISSUE THREAD

**Issue by hkouns** [bvcms/bvcms#79]
@davidcarroll ,
I am working on a couple Handlebars helpers and I have run into an issue and need some guidance.

The first I am working on is to add a Helper to duplicate the InOrg Python Event.

When I set it up like the "IfEqual" helper then it works properly:

Code
```
            /// IfInOrg in the form of:  {{IfInOrg pid orgId}}
            Handlebars.RegisterHelper("IfInOrg", (writer, options, context, args) =>
            {
                var pid = args[0];
                var orgId = args[1];

                var om = (from mm in db.OrganizationMembers
                          where mm.PeopleId == pid.ToInt()
                          where mm.OrganizationId == orgId.ToInt()
                          select mm).SingleOrDefault();

                if (om != null)
                    options.Template(writer, (object)context);
                else
                    options.Inverse(writer, (object)context);
            });
```
I can then use the following template:

```
{{IfInOrg PeopleID 48}}
   True
{{Else}}
   False
{{/IfInOrg}}
```

However, if I try to get the helper to return the string itself then I get an error:
"**Reached end of template before block expression was closed**"

Here is the code for that version:

```
         /// InOrg in the form of:  {{InOrg pid orgId}}
            Handlebars.RegisterHelper("InOrg", (writer, context, args) =>
            {
                var pid = args[0];
                var orgId = args[1];

                var om = (from mm in db.OrganizationMembers
                          where mm.PeopleId == pid.ToInt()
                          where mm.OrganizationId == orgId.ToInt()
                          select mm).SingleOrDefault();

                if (om != null)
                    writer.Write("Yes");
                else
                    writer.Write("No");
            });
```

How should I modify this helper so that it properly returns a string?  I have a couple other helpers that I would like to add... but they all follow a similar pattern.  The helper returns a string based on one of the arguments that is passed in.

_Note: In this particular case, I will probably use the "IfInOrg" version, because it gives the greatest flexibility in the template.  (based on the output, I can say True or Yes or checkmark)_

**Comment by hkouns** [comment]
**Update:**  It appears to be due to the IF statement in the Helper... If I remove it and just add a writer.Write statement for testing then it works properly.
```
            /// InOrg in the form of:  {{InOrg pid orgId}}
            Handlebars.RegisterHelper("InOrg", (writer, context, args) =>
            {
                var pid = args[0];
                var orgId = args[1];

                var om = (from mm in db.OrganizationMembers
                          where mm.PeopleId == pid.ToInt()
                          where mm.OrganizationId == orgId.ToInt()
                          select mm).SingleOrDefault();

                writer.Write(args[0].ToString());
                /// if (om != null)
                ///    writer.Write("Yes");
                /// else
                ///     writer.Write("No");
            });
```
This is the issue I have ran into on the other helpers I am working on as well.  If they contain a conditional then they cause an error when invoked in the template.  Is there a way around this?

**Comment by davidcarroll** [comment]
The helper you are writing is a block helper and as such needs to start with the #InOrg then end with the /InOrg. See if that helps. I can look more closely after that.

**Comment by hkouns** [comment]
@davidcarroll thanks for the info. I have figured out most of the bugs I had...I should be able to post my code shortly.

One thing I have noticed is that Sub-expressions do not seem to be working correctly per the information shown on the Handlebars [website](http://handlebarsjs.com/expressions.html).

Does the DotNet implementation that you are using handle subexpressions?
 ... (my understanding is the feature was added to a later release ... so maybe it wasn't included in the DotNet port?)

**For instance:**
I have created a helper called {{Now}}  which results in a string "12/15/2015"

I also have a helper called {{DateOffset date offset "offset_unit"}}
where:
offset is a positive or negative number
offset_unit is "D" or "M" or "Y"

If I call: {{DateOffset "12/15/2015" 3 "M"}}  then I get "3/15/2015"
If I call: {{DateOffset (Now) 3 "M"}}  then I get "1/0/0001"  
This tells me that the subexpression "Now" is not getting processed.

One note... I AM able to use a key name returned from SQL... For instance: BDate
If I call: {{DateOffset BDate 3 "M"}}  then I get the correct response
